citesphere/src/main/webapp/WEB-INF/views/admin/apps/details.jsp

```diff
@@ -1,12 +1,63 @@
 <%@ page pageEncoding="UTF-8" %>
 <%@ taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c"%>
 <%@ taglib prefix="form" uri="http://www.springframework.org/tags/form"%>
+<script src="<c:url value="/resources/notify/bootstrap-notify.min.js" />"></script>
+<script>
 
-<div class="pull-right" style="margin-top: 20px;">
-	<a href="<c:url value="/admin/apps/add" />" class="btn btn-primary"><i
-		class="fas fa-plus-circle"></i> Add App</a> &nbsp;&nbsp; 
-	<a href="<c:url value="/admin/apps" />" class="btn btn-primary">Show All Apps</a>
+$(function() {
+	$('#secret').click(function() {
+		$('#client-update-confirmation').modal("show");	
+		event.preventDefault();
+	});
+	
+	$("#update-client-button").click(function() {
+		var url = "/${clientId}/secret/update";
+		$.ajax({
+			'url': '<c:url value= "/admin/apps"/>'+url+"?${_csrf.parameterName}=${_csrf.token}",
+			'type': "POST",
+			'success': function(response){
+				 $('#secretKey').text(response.secret);
+				 $.notify('<i class="fas fa-check-circle"></i> New secret key successfully generated!', {
```

Comment:
should be "New secret successfully generated!" not "New secret **key** successfully generated!"